File: solutions/c/04-pn5/code/src/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
  // Flush after every printf
  setbuf(stdout, NULL);

  while (1) {
    printf("$ ");

    // Wait for user input
    char input[100];
    fgets(input, 100, stdin);

    // Remove the trailing newline
    input[strlen(input) - 1] = '\0';

    if (strcmp(input, "exit") == 0) {
      break;
    }

    printf("%s: command not found\n", input);
  }
  return 0;
}

File: solutions/dart/04-pn5/code/bin/main.dart
import 'dart:io';

void main(List<String> arguments) {
  while (true) {
    stdout.write('\$ ');

    // Wait for user input
    String command = stdin.readLineSync()!;
    if (command == 'exit') {
      break;
    }
    print('$command: command not found');
  }
}
